models/address.go

```diff
@@ -86,6 +92,13 @@ type MQTTPubAddress struct {
 	Retained       bool
 	AutoReconnect  bool
 	ConnectTimeout int
+
+	Schema string
+	// SecretPath is used to specify the secret path to store the credential(username and password) for connecting the MQTT broker
+	// User need to store the credential via the /secret API before sending the notification
+	SecretPath string
+	// AuthMode is the SMTP authentication mechanism. Currently, 'usernamepassword' is the only AuthMode supported by this service, and the secret keys are 'username' and 'password'.
```

Comment:
SMTP -> MQTT

In addition, for MQTT, I think the valid values for AuthMode should be 'usernamepassword', 'clientcert', 'cacert', and 'none'. See https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/pkg/secure/mqttfactory.go#L87-L121 for reference.

Reply:
Agree to have `clientcert`, but the `cacert` mode seems redundant.
```
case messaging.AuthModeCA:
		// Nothing to do here for this option
```

Reply:
Using `cacert` only is one of the approaches to establish an SSL/TLS connection.

The mosquito test server (https://test.mosquitto.org/, on port 8883) is a case in point.
The client only needs to verify the server certificate using the CA certificate available at https://test.mosquitto.org/ssl/mosquitto.org.crt.

Reply:
I thought the user must provide the CA cert for TLS connection, so we don't need the AuthModeCA.
https://github.com/IOTechSystems/edgex-go-private/pull/244/files#diff-8f6d4a1a57476585a2a4b4dec737e7fcfcfd62fa71a8a096da68fa5c7fa3720bR83

Reply:
If I'm not mistaken, providing a CA certificate is not a must, it depends on whether the server certificate is signed by a trusted third party (i.e. a Certificate Authority). Generally, a set of trusted CA certificates are already included in client machines/softwares.

Reply:
> whether the server certificate is signed by a trusted third party

Got it. 
- The user don't need to provide the CA cert if the cert is provided from trusted provider like letsencrypt. 
- the user need to provide the CA cert if the cert is self-signed .